lib/hammer_cli/options/option_definition.rb

```diff
@@ -104,20 +100,23 @@ def value_description
       end
 
       def default_conversion_block
-        if !value_formatter.nil?
-          value_formatter.method(:format)
-        elsif flag?
+        if flag?
           Clamp.method(:truthy?)
+        else
+          formatter = value_formatter.method(:format)
+          lambda do |value|
+            if value == (ENV['HAMMER_NIL'] || HammerCLI::Options::NIL_SUBST)
+              HammerCLI::NilValue
+            else
+              formatter.call(value)
+            end
```

Comment:
How about turning the lambda into `OptionDefinition`'s method? It seems a bit cleaner and could be more easily tested this way.

```ruby
def default_conversion_block
  if flag?
    Clamp.method(:truthy?)
  else
    self.method(:format_or_nil)
  end
end

def format_or_nil(value)
  if value == (ENV['HAMMER_NIL'] || HammerCLI::Options::NIL_SUBST)
    HammerCLI::NilValue
  else
    value_formatter.format(value)
  end
end
```

Reply:
in the example above I suppose the formatter should be format_or_nil's argument then it makes sense to me, just the name should be improved :-)

Reply:
I'm afraid the result of `default_conversion_block` is somehow used by clamp and expects only one argument which is the value. But maybe I'm wrong